winconf.pri

```diff
@@ -1,68 +1,83 @@
-# Adapt these paths on Windows
-
-# Point this to the boost include folder
-INCLUDEPATH += $$quote(C:/qBittorrent/boost_1_51_0)
-# Point this to the libtorrent include folder
-INCLUDEPATH += $$quote(C:/qBittorrent/RC_0_16/include)
-# Point this to the zlib include folder
-INCLUDEPATH += $$quote(C:/qBittorrent/Zlib/include)
-# Point this to the openssl include folder
-INCLUDEPATH += $$quote(C:/qBittorrent/openssl/include)
-
-# Point this to the boost lib folder
-LIBS += $$quote(-LC:/qBittorrent/boost_1_51_0/stage/lib)
-# Point this to the libtorrent lib folder
-LIBS += $$quote(-LC:/qBittorrent/RC_0_16/bin/path-according-to-the-build-options-chosen)
-# Point this to the zlib lib folder
-LIBS += $$quote(-LC:/qBittorrent/Zlib/lib)
-# Point this to the openssl lib folder
-LIBS += $$quote(-LC:/qBittorrent/openssl/lib)
-
-# BOOST DEFINES
-DEFINES += BOOST_ALL_NO_LIB
 DEFINES += BOOST_ASIO_HASH_MAP_BUCKETS=1021
-# Disable this if building against libtorrent 1.1.x (RC_1_1)
-DEFINES += BOOST_ASIO_SEPARATE_COMPILATION
 # After 1.55 some Windows users reported regular UI freezes.
 # This makes ASIO use the pre-1.56 way of doing things. See issue #2003
 DEFINES += BOOST_ASIO_DISABLE_CONNECTEX
-# Boost 1.60+ defaults to Vista+ support. The define below enables XP support again.
-DEFINES += BOOST_USE_WINAPI_VERSION=0x0501
 DEFINES += BOOST_EXCEPTION_DISABLE
-DEFINES += BOOST_SYSTEM_STATIC_LINK=1
 
-# LIBTORRENT DEFINES
 DEFINES += TORRENT_USE_OPENSSL
 DEFINES += TORRENT_DISABLE_GEO_IP
 DEFINES += TORRENT_DISABLE_RESOLVE_COUNTRIES
 
-# QBT DEFINES
 DEFINES += UNICODE
 DEFINES += _UNICODE
 DEFINES += WIN32
 DEFINES += _WIN32
 DEFINES += WIN32_LEAN_AND_MEAN
-DEFINES += NTDDI_VERSION=0x05010000
-DEFINES += _WIN32_WINNT=0x0501
-DEFINES += _WIN32_IE=0x0501
 DEFINES += _CRT_SECURE_NO_DEPRECATE
 DEFINES += _SCL_SECURE_NO_DEPRECATE
 DEFINES += __USE_W32_SOCKETS
 DEFINES += _FILE_OFFSET_BITS=64
 DEFINES += NOMINMAX
 
 CONFIG(debug, debug|release) {
-  DEFINES += TORRENT_DEBUG
-} else {
-  DEFINES += NDEBUG
+    DEFINES += TORRENT_DEBUG
+}
+else {
+    DEFINES += NDEBUG
 }
 
 # Enable backtrace support
 CONFIG += strace_win
 
+strace_win {
+    win32-g++* {
+        contains(QMAKE_HOST.arch, x86) {
+            # i686 arch requires frame pointer preservation
+            QMAKE_CXXFLAGS_RELEASE += -fno-omit-frame-pointer
+            QMAKE_CXXFLAGS_DEBUG += -fno-omit-frame-pointer
+        }
+
+        QMAKE_LFLAGS += -Wl,--export-all-symbols
+
+        LIBS += libdbghelp
+    }
+    else {
+        contains(QMAKE_HOST.arch, x86) {
+            # i686 arch requires frame pointer preservation
+            QMAKE_CXXFLAGS_RELEASE += -Oy-
+            QMAKE_CXXFLAGS_DEBUG += -Oy-
+        }
+
+        release {
```

Comment:
I'm confused by this line.
Isn't it should be `CONFIG(release, debug|release)`?

Reply:
You can simply try it, put `message(im in)` into this block, i tried it and that block is doing nothing, is executed in debug and in release too.

Anyway, `QMAKE_CXXFLAGS_RELEASE += -Zi` this flag is used only in release build, so it doesn't matter.
Question is, if we want to generate debug info for debug and release builds, it is described [here](https://docs.microsoft.com/en-us/cpp/build/reference/debug-generate-debug-info)

We have to options, call it like this, without the release block:
```
QMAKE_CXXFLAGS_RELEASE += -Zi
QMAKE_LFLAGS += "/DEBUG"
```

Or:
```
QMAKE_CXXFLAGS_RELEASE += -Zi
QMAKE_LFLAGS_RELEASE += "/DEBUG"
```

And here is the scope syntax http://doc.qt.io/qt-5/qmake-language.html#scopes I think that `release {` block should be called only in release builds.

Reply:
This `release {` block was added by @sledgehammer999 in `git show 1f0d4ce871292199b4c1ea082ef14f27a96fc1ba`, this block was intended to run only in release build by comments, I checked it in deepgit :), but @sledgehammer999 is more relevant to tell :)

Reply:
QMAKE_CXXFLAGS_RELEASE, QMAKE_LFLAGS_RELEASE used now instead of `release` scope.